src/charm.py

```diff
@@ -71,6 +86,10 @@ def __init__(self, *args):
         self.framework.observe(self.on.ingress_relation_joined, self._configure)
         self.framework.observe(self.on.ingress_relation_changed, self._configure)
         self.framework.observe(self.on.ingress_relation_broken, self._configure)
+        self.framework.observe(self.on.receive_remote_write_relation_created, self._configure)
+        self.framework.observe(self.on.receive_remote_write_relation_broken, self._configure)
+        self.framework.observe(self.on.prometheus_peers_relation_joined, self._configure)
+        self.framework.observe(self.on.prometheus_peers_relation_departed, self._configure)
```

Comment:
@mmanciop is going to have lots of fun borking multi unit prom deployments

Reply:
Probably! In the remote_write case multi-unit will put the app in a blocked state.